### PR TITLE
Related doc handling

### PR DESCRIPTION
**Pull request by SebCorbin**

### CODE REVIEW

File: terracommon/datastore/fields.py [hunk 1]
```diff
@@ -38,13 +42,11 @@ def to_internal_value(self, data):
 
         else:
             try:
-
                 decoded = base64.b64decode(encoded)
```
Comment (nahuelange) [inline terracommon/datastore/fields.py]: This seems to be robed from https://github.com/Terralego/terra-back/pull/18/files ?

File: terracommon/datastore/models.py [hunk 2]
```diff
@@ -32,9 +34,10 @@ class Meta:
 
 
 def related_document_path(instance, filename):
+    filename = Path(filename)
     return (f'documents/'
             f'{instance.content_type.app_label}_{instance.content_type.model}'
-            f'/{instance.object_id}/{instance.key}')
+            f'/{instance.object_id}/{instance.key}{filename.suffix}')
```
Comment (nahuelange) [inline terracommon/datastore/models.py]: What if there is no suffix? will it be suffixed by None ?

File: terracommon/tropp/serializers.py [hunk 3]
```diff
@@ -162,6 +163,20 @@ def update(self, instance, validated_data):
             picture_ids = [p.pk for p in validated_data['pictures']]
             instance.pictures.exclude(pk__in=picture_ids).delete()
 
+        # Handle related documents
+        related_docs = validated_data.pop('related')
+        # Remove stale
+        instance.related.exclude(key__in=[r['key'] for r in related_docs]).delete()
+        for related in related_docs:
+            file = related['document']
+            file.name = f"{related['key']}.{file.content_type.split('/')[1]}"
```
Comment (nahuelange) [inline terracommon/tropp/serializers.py]: Is this part enough error-proof ?

File: terracommon/datastore/models.py (hunk 2 above)
Comment (SebCorbin) [reply terracommon/datastore/models.py]:
```
>>> from pathlib import Path
>>> Path('prout').suffix
''
```

File: terracommon/datastore/fields.py (hunk 1 above)
Comment (SebCorbin) [reply terracommon/datastore/fields.py]: Yes, or the inverse :)

**Comment by SebCorbin** [comment]
Closed in favor of https://github.com/Terralego/terra-opp/pull/2

**Pull request closed without merging**